src/Components/Aspire.Keycloak/AspireKeycloakExtensions.cs

```diff
@@ -0,0 +1,117 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Keycloak;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Extensions.Hosting;
+
+/// <summary>
+/// Provides extension methods for registering Keycloak-related services in an <see cref="IHostApplicationBuilder"/>.
+/// </summary>
+public static class AspireKeycloakExtensions
+{
+    private const string DefaultConfigSectionName = "Aspire:Keycloak";
+    private const string KeycloakBackchannel = nameof(KeycloakBackchannel);
+
+    /// <summary>
+    /// Adds Keycloak JWT Bearer authentication to the application.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="connectionName">The connection name to use to find a connection string.</param>
+    /// <param name="authenticationScheme">The authentication scheme name. Default is "Bearer".</param>
+    /// <param name="configureJwtBearerOptions">An optional action to configure the <see cref="JwtBearerOptions"/>.</param>
+    /// <param name="configureSettings">An optional action to configure the <see cref="KeycloakSettings"/>.</param>
+    public static void AddKeycloakJwtBearer(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        string authenticationScheme = JwtBearerDefaults.AuthenticationScheme,
+        Action<JwtBearerOptions>? configureJwtBearerOptions = null,
+        Action<KeycloakSettings>? configureSettings = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        var settings = new KeycloakSettings();
+        builder.Configuration.GetSection(DefaultConfigSectionName).Bind(settings);
+
+        if (builder.Configuration.GetConnectionString(connectionName) is string connectionString)
+        {
+            settings.ParseConnectionString(connectionString);
+        }
```

Comment:
It's not clear to me why there's connection string logic here? Wouldn't the OIDC URL be obtained via Service Discovery?

Reply:
Sorry, I don't know how to retrieve the Keycloak endpoint if not provided via the connection string. Any example of how to do it via just service discovery?

I did it this way because then the consuming projects don't even need to know where the server lives. We just grab the conn string here, and extract the endpoint from it.

Reply:
Switched to service discovery.